NDB.Covid19/NDB.Covid19.Droid/Views/InfectionStatus/InfectionStatusActivity.cs

```diff
@@ -82,11 +81,18 @@ protected override void OnCreate(Bundle savedInstanceState)
                 OnMessageStatusChanged);
             MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_UPDATE_DAILY_NUMBERS, OnAppDailyNumbersChanged);
 
+            if(GetIsAppLaunchedToPullKeys())
+            {
+                UpdateKeys();
+                SetIsAppLaunchedToPullKeys(false);
+            }
+            
             CheckIfShowBackgroundActivityDialog();
 
             if (GetIsBackgroundActivityDialogShowEnableNewUser() == false
                 && GetIsBackgroundActivityDialogShowEnable()
-                && BatteryOptimisationUtils.CheckIsEnableBatteryOptimizations() == false)
+                && BatteryOptimisationUtils.CheckIsEnableBatteryOptimizations() == false
```

Comment:
do not mix "== true/false "syntax with direct calls to methods returning bools. so add "== true" to GetIsBackgroundActivityDialogShowEnable and GetIsAppLaunchedToShowDialog, or replace GetIsBackgroundActivityDialogShowEnableNewUser() == false with !GetIsBackgroundActivityDialogShowEnableNewUser(), and BatteryOptimisationUtils.CheckIsEnableBatteryOptimizations() with !BatteryOptimisationUtils.CheckIsEnableBatteryOptimizations()

Reply:
Ok, I will correct that.